### PR TITLE
DM-44793: Get amp-to-amp offset pedestals from ISR metadata into Chronograf

### PR DESCRIPTION
**Pull request by enourbakhsh**

### CODE REVIEW

File: python/lsst/ip/isr/ampOffset.py [hunk 1]
```diff
@@ -235,16 +235,21 @@ def run(self, exposure):
...
             pedestals = np.nan_to_num(np.linalg.lstsq(A, B, rcond=None)[0])
 
-        metadata = exposure.getMetadata()
+        metadata = exposure.getMetadata()  # Exposure metadata.
+        self.metadata["AMPOFFSET_PEDESTALS"] = {}  # Task metadata.
         for amp, pedestal in zip(amps, pedestals):
             ampIm = exposure.image[amp.getBBox()].array
             ampIm -= pedestal
             ampName = amp.getName()
+            # Add the amp pedestal to the exposure metadata.
```
Comment (enourbakhsh) [inline python/lsst/ip/isr/ampOffset.py:244]: Note that I didn't remove the amp offset pedestals from the exposure metadata after adding them to the task metadata.

File: python/lsst/ip/isr/ampOffset.py [hunk 2]
```diff
@@ -235,16 +235,21 @@ def run(self, exposure):
             # ensuring that no values are erroneously added/subtracted.
             pedestals = np.nan_to_num(np.linalg.lstsq(A, B, rcond=None)[0])
 
-        metadata = exposure.getMetadata()
+        metadata = exposure.getMetadata()  # Exposure metadata.
+        self.metadata["AMPOFFSET_PEDESTALS"] = {}  # Task metadata.
```
Comment (natelust) [inline python/lsst/ip/isr/ampOffset.py:239]: May I suggest a multiple binding here to be something like:
```ampOffsetPedestals = self.metadata["AMPOFFSET_PEDESTALS"] = {}
...
ampOffsetPedestals[ampName] = float(pedestal)

Comment (enourbakhsh) [reply python/lsst/ip/isr/ampOffset.py]: Done!

Comment (enourbakhsh) [reply python/lsst/ip/isr/ampOffset.py]: Ah, I ran some tests and found that multiple binding doesn't work with `TaskMetadata` the way we expect, i.e., this gives us empty results:
```python
        # Exposure metadata.
        metadata = exposure.getMetadata()
        # Task metadata.
        ampOffsetMetadata = self.metadata["AMPOFFSET_PEDESTALS"] = {}
        for amp, pedestal in zip(amps, pedestals):
            ampIm = exposure.image[amp.getBBox()].array
            ampIm -= pedestal
            ampName = amp.getName()
            # Add the amp pedestal to the exposure metadata.
            metadata.set(
                f"LSST ISR AMPOFFSET PEDESTAL {ampName}",
                float(pedestal),
                f"Pedestal level subtracted from amp {ampName}",
            )
            # Add the amp pedestal to the "Task" metadata as well.
            # Needed for Sasquatch/Chronograf!
            ampOffsetMetadata[ampName] = float(pedestal)
```
Debugging result:
```python
ipdb> ampOffsetMetadata
{'0': -0.20423207279935987, '1': -0.06727251761094916, '2': -0.28773895061785293, '3': 0.5592435410281624}
ipdb> self.metadata["AMPOFFSET_PEDESTALS"]
TaskMetadata(scalars={}, arrays={}, metadata={})
```

I'm reverting it.

**Review by leeskelvin — APPROVED** [review]
This LGTM, thanks!

**Pull request merged**